### ISSUE TITLE
Tensorflow lite model requests a buffer bigger than the neccesary

### ISSUE THREAD

**Issue by rockbass2560** [tensorflow/tensorflow#26199]
Hi, I created a custom model using keras in tensorflow. The version that I used was tensorflow nightly 1.13.1. I used the official tool to build the tensorflow lite model (the method tf.lite.TFLiteConverter.from_keras_model_file ).

After I created the model I reviewed the input shape and nothing seems is bad.

The input and output shapes in tensorflow lite model are:

[{'name': 'input_1', 'index': 59, 'shape': array([  1, 240, 240,   3], dtype=int32), 'dtype': , 'quantization': (0.0, 0)}]

[{'name': 'dense/Softmax', 'index': 57, 'shape': array([1, 6], dtype=int32), 'dtype': , 'quantization': (0.0, 0)}]

you can note that input shape is 1 * 240 * 240 * 3 so I expected that the buffer would have a size of 172800 units.

However, when I try to run the model in an android device I received the next error:

E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.megacode, PID: 15067
    java.lang.RuntimeException: Unable to create application com.megacode.base.ApplicationBase: java.lang.IllegalArgumentException: Cannot convert between a TensorFlowLite buffer with 691200 bytes and a ByteBuffer with 172800 bytes.
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5771)
        at android.app.ActivityThread.-wrap2(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1648)
I don't understand the reason why the model request an input shape of 691200 units.

If someone has a suggestion I would appreciate it

**Comment by jdduke** [comment]
The input type is int32, which is 4 bytes (4 x 172800 == 691200).

**Comment by charlizesmith** [comment]
Did you get any solution for above problem? I am facing same issue.

**Comment by rockbass2560** [comment]
In my case I needed to put the size in bytes of int type. So the operation was 4x240x240x3

**Comment by charlizesmith** [comment]
Thanks rockbass2560, i tried and in my case it is issue of input size. it is 224 works for me. I am not getting this error and model run successfully but not getting image mask. How to configure remaining value in the below formula?

> ByteBuffer.allocateDirect(1 * imageSize * imageSize * NUM_CLASSES * 4)
